src/Traits/ManagerRepository.php

```diff
@@ -113,14 +113,21 @@ protected function doFindIdsBy(array $criteria, array $orderBy = null, $limit =
         if ($keys === [$idField] && is_null($orderBy) && is_null($limit) && is_null($offset)) {
             return is_array($criteria[$idField]) ? $criteria[$idField] : [$criteria[$idField]];
         }
-        $qb = $this->getEntityManager()
-            ->createQueryBuilder()
-            ->select("x.${idField}")
-            ->distinct()
-            ->from($this->getEntityName(), 'x');
+        $fields = [$idField];
+        if (is_array($orderBy)) {
+            $fields = array_unique([...$fields, ...array_keys($orderBy)]);
+        }
+        $dqlSelect = implode(', ', array_map(function ($field) {
+            return "x.${field}";
+        }, $fields));
```

Comment:
```suggestion
        $dqlSelect = implode(', ', array_map(fn ($field) => "x.${field}", $fields));
```
I like it better without the line break, but I'm ok with this too.